dicodile/_dicodile.py

```diff
@@ -9,18 +9,87 @@
-
-    lmbd_max = get_lambda_max(X, D_hat).max()
+DEFAULT_DICOD_KWARGS = dict(max_iter=int(1e8), n_seg='auto',
+                            strategy='greedy', timeout=None)
+
+
+def dicodile(X, D_init, reg=.1, n_iter=100, eps=1e-5, window=False,
+             z_positive=True, n_workers=4, w_world='auto',
+             tol=1e-3, hostfile=None, dicod_kwargs={},
+             random_state=None, verbose=0):
+    r"""Convolutional dictionary learning.
+
+    Computes a sparse representation of a signal X, returning a dictionary
+    D and a sparse activation signal Z such that X is close to
+    :math:`Z \ast D`.
+
+    This is done by solving the following optimization problem:
+
+    .. math::
+        \underset{Z,D, \left \| D_{k}\right \|\leq 1}{min} \frac{1}{2}
+        \left \| X - Z \ast D \right\|_{2}^{2} +
+        \lambda \left \| Z \right \|_{1}
+
+    The support for X is noted sig_support
+    The support for D is noted atom_support
+
+    Parameters
+    ----------
+    X : ndarray, shape (n_channels, *sig_support)
+        Signal to encode
+    D_init : ndarray, shape (n_atoms, n_channels, *atom_support)
+        Current atoms dictionary
+    reg : float, defaults to .1
+        Regularization parameter, in [0,1]
+        The λ parameter is computed as reg * lambda_max
+    n_iter : int, defaults to 100
+        Maximum number of iterations
+    eps : float, defaults to 1e-5
+        Tolerance for the stopping criterion. A lower value will result in
+        more computing time.
+    window : bool
+        Apply window on dictionary
+    z_positive : bool, default True
+        if True, adds a constraint that the activations Z must be positive
+    n_workers : int, defaults to 4
+        Number of workers used to compute the convolutional sparse coding
+        solution.
+    w_world : int or {{'auto'}}
+        Number of jobs used per row in the splitting grid. This should divide
+        n_workers.
+    tol : float, defaults to 1e-3
+        Tolerance for minimal update size
+    hostfile : str or None
+        MPI hostfile as used by `mpirun`. See your MPI implementation
+        documentation. Defaults to None.
+    dicod_kwargs : dict
+        Extra arguments passed to the dicod funtion.
+        See `dicodile.update_z.dicod`
+    random_state : None or int or RandomState
+        Random state to seed the random number generator.
+    verbose : int, defaults to 0
+        Verbosity level, higher is more verbose
+
+    Returns
+    -------
+    pobj : list of float
+        list of costs
+    times : list of float
+        list of running times (seconds) for each dictionary
+        and activation update step.
+        The total running time of the algorithm is given by
+        sum(times)
+    D_hat : ndarray, shape (n_channels, *sig_support)
+        Updated atoms dictionary.
+    Z_hat : ndarray, shape (n_channels, *valid_support)
+        Activations of the different atoms
+        (where or when the atoms are estimated).
```

Comment:
maybe add See also section for dicod function?

Reply:
I feel pobj and times should be returned last. Most users don't care about these.

Reply:
I agree

Reply:
actually, I think it would be nice to even not return these by default?
When we update the API, we can think of adding a `return_cost` argument that would trigger the construction and return of these 2 objects only if needed.